src/Invictus.Testing/LogicAppClient.cs

```diff
@@ -0,0 +1,484 @@
+        /// Runs the current logic app resource using the given <paramref name="triggerName"/>.
+        /// </summary>
+        /// <param name="triggerName">The name of the trigger that executes a workflow in the logic app.</param>
+        public async Task RunAsync(string triggerName = "")
+        {
+            if (String.IsNullOrEmpty(triggerName))
+            {
+                triggerName = await GetTriggerNameAsync();
+            }
+
+            _logger.LogTrace("Run the workflow trigger of logic app '{LogicAppName}' in resource group '{ResourceGroup}'", _logicAppName, _resourceGroup);
+            await _logicManagementClient.WorkflowTriggers.RunAsync(_resourceGroup, _logicAppName, triggerName);
+        }
+
+        
+        /// <summary>
+        /// Gets the logic app definition information.
+        /// </summary>
+        public async Task<LogicApp> GetMetadataAsync()
+        {
+            Workflow workflow = await _logicManagementClient.Workflows.GetAsync(_resourceGroup, _logicAppName);
+            return Converter.ToLogicApp(workflow);
+        }
+
+        /// <summary>
+        /// Run logic app on the current trigger URL, posting the given <paramref name="headers"/>.
+        /// </summary>
+        /// <param name="headers">The headers to send with the trigger URL of the current logic app.</param>
+        public async Task TriggerAsync(IDictionary<string, string> headers)
+        {
+            Guard.NotNull(headers, nameof(headers));
+            Guard.NotAny(headers, nameof(headers));
+
+            LogicAppTriggerUrl triggerUrl = await GetTriggerUrlAsync();
+
+            _logger.LogTrace("Trigger the workflow of logic app '{LogicAppName}' in resource group '{ResourceGroup}'", _logicAppName, _resourceGroup);
+            using (var request = new HttpRequestMessage(HttpMethod.Post, triggerUrl.Value))
+            {
+                foreach (KeyValuePair<string, string> header in headers)
+                {
+                    request.Headers.Add(header.Key, header.Value);
+                }
+
+                await HttpClient.SendAsync(request);
+            }
+        }
+
+        /// <summary>
+        /// Gets the URL on which the workflow with the <paramref name="triggerName"/> can be run.
+        /// </summary>
+        /// <param name="triggerName">The name of the trigger that relates to a workflow.</param>
+        public async Task<LogicAppTriggerUrl> GetTriggerUrlAsync(string triggerName = "")
+        {
+            if (String.IsNullOrEmpty(triggerName))
+            {
+                triggerName = await GetTriggerNameAsync();
+            }
+            
+            _logger.LogTrace("Request the workflow trigger URL of logic app '{LogicAppName}' in resource group '{ResourceGroup}'", _logicAppName, _resourceGroup);
+            WorkflowTriggerCallbackUrl callbackUrl = 
+                await _logicManagementClient.WorkflowTriggers.ListCallbackUrlAsync(_resourceGroup, _logicAppName, triggerName);
```

Comment:
Just to be sure, will this work when no triggerName has been passed along?
What I mean is, won't the ListCallBackUrlAsync-method check for a trigger named "", which won't exist?

Reply:
Hmm, this was just as-is moved. When the trigger name is passed along as empty or null the name of the first trigger on the app is retrieved, but I left a Todo comment in that method bc we alloi indeed an empty string to be the return value when no trigger can be found.

Can we throw here? Also, should we split this in two methods with maybe a slightly different me so the consumer knows it finds the first trigger in the background?

Reply:
In fact a logic app will only have a single trigger, but I do agree that the current use isn't clear enough. Let's split it up in 2 methods as you suggest.

Reply:
Ok, I've done that, and also updated the `RunAsync` since this was set up in the same manner.